sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -148,6 +144,28 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
   override def hashCode(): Int = getClass.hashCode()
 
   override def equals(other: Any): Boolean = other.isInstanceOf[JsonFileFormat]
+
+  override def validateDataType(dataType: DataType, isReadPath: Boolean): Unit = dataType match {
```

Comment:
how about

```
the base class
def validateDataType(dataType: DataType, isReadPath: Boolean): Boolean = dataType match {
  case BooleanType | ByteType | ShortType | IntegerType | LongType | FloatType | DoubleType |
    StringType | BinaryType | DateType | TimestampType | _: DecimalType => true
  case _ => false
}

json
override def validateDataType(dataType: DataType, isReadPath: Boolean): Boolean = {
  case st: StructType => st.forall { f => validateDataType(f.dataType, isReadPath) }
  case ArrayType...
  ...
  case other => super.validateDataType(other)
}


```

Reply:
Then the base class could break other existing file formats, right?

Reply:
what do you mean by break? If people use internal API(like avro), they are responsible to update their code for the internal API changes in new Spark releases.